docs/enhancements/cluster-deployment-condition-refactor.md

```diff
@@ -0,0 +1,95 @@
+# Hive ClusterDeployment Conditions Refactor
+
+## Summary
+
+Reduce the number of conditions on ClusterDeployment and provide callers better options to monitor progress.
+
+## Motivation
+
+ClusterDeployment now carries over 20 conditions by default, from a variety of controllers. This is difficult for users to parse and there is not presently one condition that can be monitored for overall progress of an install.
+
+## Goals
+
+  1. Reduce the number of conditions on ClusterDeployment.
+  1. Provide callers high level conditions they can monitor.
+
+
+## Background
+
+Current 22 conditions:
+
+- Unreachable
+- ClusterImageSetNotFound
+- ControlPlaneCertificateNotFound
+- Hibernating
+- RelocationFailed
+- SyncSetFailed
+- AWSPrivateLinkFailed
+- AWSPrivateLinkReady
+- ActiveAPIURLOverride
+- AuthenticationFailure
+- ClusterInstallCompleted
+- ClusterInstallFailed
+- ClusterInstallRequirementsMet
+- ClusterInstallStopped
+- DNSNotReady
+- DeprovisionLaunchError
+- IngressCertificateNotFound
+- InstallImagesNotResolved
+- InstallLaunchError
+- InstallerImageResolutionFailed
+- ProvisionFailed
+- ProvisionStopped
+
+## Proposal
+
+Focusing only on the 12 conditions controlled by the clusterdeployment controller:
+
+- ClusterImageSetNotFound - Move to RequirementsMet or possibly ClusterInstallRequirementsMet.
+- AuthenticationFailure - Move to RequirementsMet or ClusterInstallRequirementsMet.
+- ClusterInstallCompleted - Merge into Installed.
+- ClusterInstallFailed - Merge into ProvisionFailed.
+- ClusterInstallRequirementsMet - Merge into RequirementsMet.
+- ClusterInstallStopped - Merge into ProvisionStopped.
+- DNSNotReady - Move to RequirementsMet
+- DeprovisionLaunchError -  Rename to DeprovisionError as a general bucket for any deprovision problems we can encounter.
+- InstallLaunchError - Move to RequirementsMet or ProvisionFailed
+- InstallerImageResolutionFailed - Move to RequirementsMet or ProvisionFailed
```

Comment:
As far as definition is concerned, there seems to be an overlap of `InstallerImageResolutionFailed` with `InstallImagesNotResolved`... I think both belong in `ProvisionFailed`